Heatmap api

jonty-dawson opened the pull request.

jonty-dawson committed:
dynamic heatmaps - initial commit, boilerplate from PolygonShape

- plan to factor out common code between Heatmap + Polygon once further
advanced

Buddy: Sam

jonty-dawson committed:
heatmap api wip

- Add WeightedLatLngAlt (geographic coord with arbitrary associated scalar)
- Options aggregates polygon options, add heatmap-specific options

Buddy: Malcolm

jonty-dawson committed:
add opacity to heatmaps

Buddy: Malcolm

jonty-dawson committed:
heatmap api continued

- further params modelled by native api exposed in Java api

Buddy: Ali

jonty-dawson committed:
add color gradient to heatmap

Buddy: Malcolm

jonty-dawson committed:
add gradient to Heatmap, with mutator

Buddy: Mark

jonty-dawson committed:
add mutation of Heatmap data

+ rename data->weightedPoints

Buddy: Micky

jonty-dawson committed:
IntensityBias now passed explicity via api

also fix double vs float issues

Buddy: Michael C

jonty-dawson committed:
add mutation of heatmap radii, texture resolution

Buddy: Sam

jonty-dawson committed:
fix param validation - allow empty polygon verts

Buddy: John B

jonty-dawson committed:
expose texture border via api

Buddy: Micky

jonty-dawson committed:
fix uninitialised field

jonty-dawson committed:
Generalise heatmap api to allow 1..N heatmap intensity textures to be generated

 - previously always 2
 - with large differences between radii, interpolation artefacts become aparent. This change allows additional intensity textures with different radii to be generated, enabling greater fidelity of resultant visuals.

Buddy: Micky

jonty-dawson committed:
add toggle to use approximation of gaussian blur when computing heatmap,
defualt true

Buddy: Sam

jonty-dawson committed:
separate out jni state mutation methods

Buddy: Michael C

jonty-dawson committed:
split out native update of intensity bias + scale (intensity texture
generation now depends on bias)

Buddy: Michael C

jonty-dawson committed:
reorder / rename *StartParams -> *Stops for consistency across apis
(wrld.js)

Buddy: Sam

jonty-dawson committed:
further rename

jonty-dawson committed:
add additional scaling parameter per density texture map

- api renames to follow concept of interpolating between 1..N density maps, each built with a specific splat radius and gain

Buddy: Malcolm

jonty-dawson committed:
add option to interpolate between heatmap density maps based on camera
zoom

- this added as performing entirely in app proved difficult, some camera
movements do not fire event (eg indoor map slider control pressed causes
camera zoom, but no notification via camera api).

Buddy: Micky

jonty-dawson committed:
add enum HeatmapOcclusionMapFeature, use array rather than bitfield in
public api

- native interop still bitfield

Buddy: Micky

jonty-dawson committed:
Merge branch 'master' into heatmap_api

jonty-dawson committed:
additional param validation

- addresses point raised in native code review https://github.com/wrld3d/eegeo-mobile/pull/9#pullrequestreview-228649384
- in setOptions, avoid redundant native update (potentially costly, causing unnecessary rebuilding of view components)

Buddy: Malcolm

Michael-ODonnell commented on sdk/src/main/java/com/eegeo/mapapi/geometry/WeightedLatLngAlt.java:
```diff
@@ -0,0 +1,46 @@
+            return false;
+        }
+        WeightedLatLngAlt other = (WeightedLatLngAlt)(object);
+        if (Double.compare(this.intensity, other.intensity) == 0) {
```
This seems to be returning false if intensitys are equal.

jonty-dawson replied:
thanks, fixed with c0c85ccf24e53b6567523d6b0c006be3e931cb52

jonty-dawson committed:
fix comparison

- addresses https://github.com/wrld3d/android-api/pull/13/#pullrequestreview-228730787

Buddy: Michael C

jonty-dawson committed:
add javadoc for heatmap api

- some renames + reordering of fields / getters / setters

Buddy: John B

jonty-dawson commented:
squash merged with a21e067017b78bd5e86d1895191eaa339595e6a1
